crates/uv-resolver/src/requires_python.rs

```diff
@@ -362,8 +362,9 @@ impl RequiresPython {
                     let Some(minor) = python_tag
```

Comment:
Does this entire part make sense? In the example above, I thought cp36 was a lower-bound, so it _would_ match`==3.10.*`.

Reply:
`cpxx-<...>-<...>` as a python tag only really makes sense in combination with a matching ABI tag, `cp311-none` doesn't really make sense: why are you requiring CPython if you're not requiring its ABI? For pragmatic reasons, we're following pip (on python 3.12), `cp` is an exact python tag unless we use `abi3`, then it's a minimum version, while `py` is a lower bound:

```
pip debug --verbose | grep manylinux_2_39_x86_64
WARNING: This command is only meant for debugging. Do not use this with automation for parsing and getting these details, since the output and options of this command may change without notice.
  cp312-cp312-manylinux_2_39_x86_64
  cp312-abi3-manylinux_2_39_x86_64
  cp312-none-manylinux_2_39_x86_64
  cp311-abi3-manylinux_2_39_x86_64
  cp310-abi3-manylinux_2_39_x86_64
  cp39-abi3-manylinux_2_39_x86_64
  cp38-abi3-manylinux_2_39_x86_64
  cp37-abi3-manylinux_2_39_x86_64
  cp36-abi3-manylinux_2_39_x86_64
  cp35-abi3-manylinux_2_39_x86_64
  cp34-abi3-manylinux_2_39_x86_64
  cp33-abi3-manylinux_2_39_x86_64
  cp32-abi3-manylinux_2_39_x86_64
  py312-none-manylinux_2_39_x86_64
  py3-none-manylinux_2_39_x86_64
  py311-none-manylinux_2_39_x86_64
  py310-none-manylinux_2_39_x86_64
  py39-none-manylinux_2_39_x86_64
  py38-none-manylinux_2_39_x86_64
  py37-none-manylinux_2_39_x86_64
  py36-none-manylinux_2_39_x86_64
  py35-none-manylinux_2_39_x86_64
  py34-none-manylinux_2_39_x86_64
  py33-none-manylinux_2_39_x86_64
  py32-none-manylinux_2_39_x86_64
  py31-none-manylinux_2_39_x86_64
  py30-none-manylinux_2_39_x86_64
```